org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -965,9 +965,12 @@ protected com.sun.tools.javac.parser.Tokens.Comment processComment(int pos, int
 
 	static void addCommentsToUnit(Collection<Comment> comments, CompilationUnit res) {
 		List<Comment> before = res.getCommentList() == null ? new ArrayList<>() : new ArrayList<>(res.getCommentList());
-		comments.stream().filter(comment -> comment.getStartPosition() >= 0 && JavacCompilationUnitResolver.noCommentAt(res, comment.getStartPosition()))
-			.forEach(before::add);
-		before.sort(Comparator.comparingInt(Comment::getStartPosition));
+		before = before.stream().filter(comment -> !generated(comment)).toList();
+		if (!before.isEmpty()) {
```

Comment:
why adding thos condition here? Could it be just an extra `.filter(comment -> !generated(comment))` to the the existing stream?